treeherder/perf/data_cycling/signature_remover.py

```diff
@@ -0,0 +1,141 @@
+import logging
+from typing import List
+
+from treeherder.services.taskcluster import TaskclusterModel
+from treeherder.perf.data_cycling.max_runtime import MaxRuntime
+
+logger = logging.getLogger(__name__)
+
+RECEIVER_TEAM_EMAIL = "perftest-alerts@mozilla.com"
+RECEIVER_EMAIL = "dhunt@mozilla.com"
+
+
+class PublicSignatureRemover:
+    """
+    This class handles the removal of signatures which are (no longer)
+    associated to any data point and sends email notifications to the entire team.
+    """
+
+    TABLE_DESCRIPTION = """Perfherder removes performance data that is older than one year and in some cases even sooner, leaving behind performance signatures that aren't associated to any data point. These as well need to be removed.
+> __Here's a summary of recently deleted performance signatures:__
+---
+    """
+
+    TABLE_HEADERS = """
+| Repository | Framework | Platform | Suite | Application |
+| :---: | :---: | :---: | :---: | :---: |
+    """
+
+    def __init__(
+        self,
+        timer: MaxRuntime,
+        taskcluster_model: TaskclusterModel,
+        max_rows_allowed=None,
+        max_emails_allowed=None,
+    ):
+        self.tc_model = taskcluster_model
+        self._subject = "Summary of deleted Performance Signatures"
+        self._content = None
+        self._max_rows_allowed = max_rows_allowed or 50
+        self._max_emails_allowed = max_emails_allowed or 10
+
+        self.timer = timer
+
+    def remove_in_chunks(self, signatures):
+        emails_sent = 0
+        rows_left = self._max_rows_allowed
+        chunk_of_signatures = []
+
+        logger.warning('Removing performance signatures with missing jobs...')
```

Comment:
This comment isn't correct. We're removing signatures which don't have any data points.